### PR TITLE
Custom build sys v4.3.0

### PR DESCRIPTION
**Pull request by walkjivefly**
Contains the changes needed to generate a custom Docker build image which can create a SYS v4.3.0 image.

After this PR is merged the correct way to generate a SYS v4.3.0 image is to run the custom build workflow.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -41,7 +41,6 @@ jobs:
       if: ${{ always() }}
       run: yes | docker image prune
     - name: Push an image to DockerHub
-      run: echo Pushing "${{ github.event.inputs.image }}" ${{ github.event.inputs.version }}
       run: ./ci.sh push "${{ github.event.inputs.image }}" ${{ github.event.inputs.version }}
```
Comment (ConanMishler) [inline .github/workflows/build.yaml:44]: This seems to be a change that was meant to be in the PR $42 to enhance the BUILD IMAGE FROM TEMPLATE workflow, but it didn't make it into that PR, so you're sneaking it into this PR. ;-)  That's fine, just checking to make sure I understand what's going on...

**Review by ConanMishler — COMMENTED** [review]
Two questions:
1. Did you confirm there is a problem with the current SYS 4.3.0 image? Just wondering bc I haven't been able to reproduce the db corruption issue I was seeing previously when SYS 4.3.0 container was stopped/restarted. Sounds like @tryiou is also not able to reproduce the issue he was seeing w/ 4.3.0 container previously. Just wondering what's the motivation for creating a new SYS 4.3.0 image using the BUILD IMAGE FROM CUSTOM workflow.
2. This PR will create new a new custom `Dockerfile` in `images/syscoin` and `images/syscoin4`. Isn't there a danger one or both of those new custom `Dockerfile`s will be overwritten if/when someone runs the BUILD IMAGE FROM TEMPLATE workflow for syscoin?

File: .github/workflows/build.yaml (hunk 1 above)
Comment (walkjivefly) [reply .github/workflows/build.yaml:44]: hmmm, this one was just cleaning up a mistake I made somewhere but didn't realise at the time.

**Comment by walkjivefly** [comment]
1. The SYS 4.3 image seems to be working fine for everyone now. Previously it was rescanning the chain on every startup but there was no indication of why in the debug.log and it reported the previous shutdown as clean. I think maybe it had something to do with the EVM activation but that's pure speculation.
The reason for the custom build is SYS 4.3 requires GCC >= 8.1 and the normal build process uses only GCC 7. 
2. There's a new Dockerfile in images/syscoin4 and that's the one that should be used to create SYS images in future, using the BUILD IMAGE FROM CUSTOM workflow. This workflow omits the generate step which creates the images/coin/Dockerfile. If anyone uses the CREATE IMAGE FROM TEMPLATE for SYS they will overwrite the Dockerfile with something which won't work, at least until we switch to GCC 8+ as default for builds.
3. I built this weeks ago. I was trying to get to the end of the workflow before PR-ing all the changes. This one ended up "on the back burner", like the Raven one (which needed even more hoops to be jumped through and is waiting for the chain to sync for some confirmation trades before I PR it).

**Comment by ConanMishler** [comment]
> * The SYS 4.3 image seems to be working fine for everyone now. Previously it was rescanning the chain on every startup but there was no indication of why in the debug.log and it reported the previous shutdown as clean. I think maybe it had something to do with the EVM activation but that's pure speculation.
>   The reason for the custom build is SYS 4.3 requires GCC >= 8.1 and the normal build process uses only GCC 7.
> * There's a new Dockerfile in images/syscoin4 and that's the one that should be used to create SYS images in future, using the BUILD IMAGE FROM CUSTOM workflow. This workflow omits the generate step which creates the images/coin/Dockerfile. If anyone uses the CREATE IMAGE FROM TEMPLATE for SYS they will overwrite the Dockerfile with something which won't work, at least until we switch to GCC 8+ as default for builds.
> * I built this weeks ago. I was trying to get to the end of the workflow before PR-ing all the changes. This one ended up "on the back burner", like the Raven one (which needed even more hoops to be jumped through and is waiting for the chain to sync for some confirmation trades before I PR it).

Thanks for the detailed answers.
> If anyone uses the CREATE IMAGE FROM TEMPLATE for SYS they will overwrite the Dockerfile with something which won't work, at least until we switch to GCC 8+ as default for builds.
This situation seems less than optimal. Is there any reason we can't immediately switch to using GCC 8+ ?

**Comment by walkjivefly** [comment]
> > * The SYS 4.3 image seems to be working fine for everyone now. Previously it was rescanning the chain on every startup but there was no indication of why in the debug.log and it reported the previous shutdown as clean. I think maybe it had something to do with the EVM activation but that's pure speculation.
> >   The reason for the custom build is SYS 4.3 requires GCC >= 8.1 and the normal build process uses only GCC 7.
> > * There's a new Dockerfile in images/syscoin4 and that's the one that should be used to create SYS images in future, using the BUILD IMAGE FROM CUSTOM workflow. This workflow omits the generate step which creates the images/coin/Dockerfile. If anyone uses the CREATE IMAGE FROM TEMPLATE for SYS they will overwrite the Dockerfile with something which won't work, at least until we switch to GCC 8+ as default for builds.
> > * I built this weeks ago. I was trying to get to the end of the workflow before PR-ing all the changes. This one ended up "on the back burner", like the Raven one (which needed even more hoops to be jumped through and is waiting for the chain to sync for some confirmation trades before I PR it).
> 
> Thanks for the detailed answers.
> 
> > If anyone uses the CREATE IMAGE FROM TEMPLATE for SYS they will overwrite the Dockerfile with something which won't work, at least until we switch to GCC 8+ as default for builds.
> 
> This situation seems less than optimal. Is there any reason we can't immediately switch to using GCC 8.1+ as the default compiler for image builds? Maybe @atcsecure has an opinion on that.

Changing toolchain is not to be undertaken lightly. It's possible (but probably unlikely) that some coins may not build with GCC 8+ without modifications. We could switch to GCC 8 and keep our fingers crossed ;-)

I'll do a few tests and see if any of the coins near the top of my todo list fail to build with GCC 8.

**Comment by ConanMishler** [comment]
> I'll do a few tests and see if any of the coins near the top of my todo list fail to build with GCC 8.

Yes, I feel it will still be good to upgrade to GCC 8 if it doesn't impact image building for any of our most important coins. I think the more we can use the CREATE IMAGE FROM TEMPLATE workflow and avoid the BUILD IMAGE FROM CUSTOM workflow, the better off we'll be.

**Pull request merged**